lib/glific/communications/wa_group_message.ex

```diff
@@ -69,9 +73,112 @@ defmodule Glific.Communications.GroupMessage do
       log_error(message, "Could not send message to contact: Check maytapi Setting")
   end
 
+  @doc """
+  Callback when we receive a message from whatsapp group providers
+  """
+  @spec receive_message(map(), atom()) :: :ok | {:error, String.t()}
+  def receive_message(%{organization_id: organization_id} = message_params, type \\ :text) do
+    Logger.info(
+      "Received message: type: '#{type}', phone: '#{message_params.sender.phone}', id: '#{message_params[:bsp_id]}'"
+    )
+
+    {:ok, contact} =
+      message_params.sender
+      |> Map.put(:organization_id, organization_id)
+      |> Contacts.maybe_create_contact()
+
+    if Contacts.contact_blocked?(contact),
+      do: :ok,
+      else: do_receive_message(contact, message_params, type)
```

Comment:
I dont think we need this as contact blocked will have no effect in receiving message from them in a WA group